### PR TITLE
v1.23.0

### PR DESCRIPTION
**Pull request by emelsimsek**

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -4,6 +4,40 @@ PgBouncer changelog
 PgBouncer 1.22.x
 ----------------
 
+**2024-06-28  -  PgBouncer 1.22.2  -  "Into the new beginnings"**
```
Comment (JelteF) [inline NEWS.md:7]: Let's make the version number 1.23.0. We only do patch releases if we have only bugfixes. This release has many changes including a bunch of features, so a minor version bump makes sense.

File: NEWS.md [hunk 2]
```diff
@@ -4,6 +4,40 @@ PgBouncer changelog
...
+- Features
+  * Add support for user name maps for for `cert` and `peer` authentication
+    methods. ([#996])
+  * Add support for negotiating the postgres protocol version with the client. ([#1007])
+  * Add support for replication connections through PgBouncer. ([#876])
+  * Add support for rolling restarts when peering is configured. ([#902])
```
Comment (JelteF) [inline NEWS.md:13]: This change included a breaking change, which should be listed at the top of the changelog. I suggest to move the whole:

SIGTERM doesn't cause immediate shutdown of the PgBouncer process anymore. It now does a "super safe shutdown": waiting for all clients to disconnect before shutting down. If you used SIGTERM to trigger an immediate shutdown in your Dockerfile or SystemD service file, you should replace that signal with a SIGQUIT to keep the existing behaviour. The new SIGTERM behaviour allows rolling restarts of multiple PgBouncer processes behind a load balancer, or listening on the same port using `so_reuseport`. ([#902])

File: NEWS.md [hunk 3]
```diff
@@ -4,6 +4,40 @@ PgBouncer changelog
...
+
+- Changes
+  * Improve `SHOW USERS` output listing the connections. ([#1040])
+  * Allow `pool_size` configuration per user. ([#1049])
+  * Allow `server_lifetime` configuration per database. ([#1057])
+  * Add support for listing dynamically created users in the output of `SHOW USERS`. ([#1052])
+  * Add support for `all` address type in hba configuration. ([#1078])
+  * Add support for automatically restarting when using systemd.  ([#1080])
```
Comment (JelteF) [inline NEWS.md:21]: Double space by accident?
```suggestion
  * Add support for automatically restarting when using systemd. ([#1080])
```

File: NEWS.md [hunk 4]
```diff
@@ -4,6 +4,40 @@ PgBouncer changelog
 PgBouncer 1.22.x
 ----------------
 
+**2024-06-28  -  PgBouncer 1.22.2  -  "Into the new beginnings"**
+- Features
+  * Add support for user name maps for for `cert` and `peer` authentication
+    methods. ([#996])
+  * Add support for negotiating the postgres protocol version with the client. ([#1007])
```
Comment (JelteF) [inline NEWS.md:11]: nit: I would move this to fixes

File: NEWS.md [hunk 5]
```diff
@@ -4,6 +4,40 @@ PgBouncer changelog
 PgBouncer 1.22.x
 ----------------
 
+**2024-06-28  -  PgBouncer 1.22.2  -  "Into the new beginnings"**
+- Features
+  * Add support for user name maps for for `cert` and `peer` authentication
+    methods. ([#996])
```
Comment (JelteF) [inline NEWS.md:10]: I think you can expand a little bit more on this (this is one of the major features of the release). Something like: This allows a lot more freedom for client-side TLS certificate authentication.

**Review by JelteF — CHANGES_REQUESTED** [review]
Two things are missing:
1. PR #727 is missing in the changelog
2. Drop support for c-ares version 1.9.0

File: NEWS.md (hunk 2 above)
Comment (emelsimsek) [reply NEWS.md:13]: Should we have a BREAKING CHANGES section?

Comment (JelteF) [reply NEWS.md]: Yeah, I think that would be good. No need for ALL CAPS though.

**Comment by eulerto** [comment]
- #1034 should be mentioned in the Fixes section.
- Why is there a fix for CI in the release commit? It should be a separate patch.
- We used to have **Features**, **Fixes** and **Cleanups** as sections. We recently introduced **Deprecations** and **Changes**. It seems the goal of Changes is to break Features into Important Features and Regular Features. It can cause some confusion IMO and I prefer to have only one section: **Features**. I'm not sure about this new section (Breaking Changes), I would put it into Features with the same proposed sentences. The "This is a minor breaking change" is clear that this commit introduces a different behavior.
- s/SystemD/systemd/
- various commits are not mentioned in the release notes. It should be in Fixes or Cleanups such as #1036 #1045 #1046 #1047 #1051 #1073 #1082 #1083 #1091.

**Comment by JelteF** [comment]
> * [Add outstanding request for auth_query #1034](https://github.com/pgbouncer/pgbouncer/pull/1034) should be mentioned in the Fixes section.
> 
>     * Why is there a fix for CI in the release commit? It should be a separate patch.

Agreed, this is merged separately now.

>     * We used to have **Features**, **Fixes** and **Cleanups** as sections. We recently introduced **Deprecations** and **Changes**. It seems the goal of Changes is to break Features into Important Features and Regular Features. It can cause some confusion IMO and I prefer to have only one section: **Features**. I'm not sure about this new section (Breaking Changes), I would put it into Features with the same proposed sentences. The "This is a minor breaking change" is clear that this commit introduces a different behavior.

I think people easily skip over a release note item description if after the first sentence they think it doesn't impact them. So I proposed to call it out a bit more heavily, especially since I don't think we make breaking changes often/ever. Keeping it in features and making **minor breaking change** bold would also be fine by me. 

>     * various commits are not mentioned in the release notes. It should be in Fixes or Cleanups such as [#1035 - patch to indicate pem certificates / keys are usable by pgbouncer #1036](https://github.com/pgbouncer/pgbouncer/pull/1036) [Fix typos #1045](https://github.com/pgbouncer/pgbouncer/pull/1045) [Silence deprecation warnings of c-ares #1046](https://github.com/pgbouncer/pgbouncer/pull/1046) [Fix tests on non US datestyle configured systems #1047](https://github.com/pgbouncer/pgbouncer/pull/1047) [Improves README #1051](https://github.com/pgbouncer/pgbouncer/pull/1051) [The 1st argument of calloc should be 'nelem'. #1073](https://github.com/pgbouncer/pgbouncer/pull/1073) [Fix minor details in config.md #1082](https://github.com/pgbouncer/pgbouncer/pull/1082) [Improve documentation on connecting to AWS RDS and SCRAM #1083](https://github.com/pgbouncer/pgbouncer/pull/1083) [Improve run python function #1091](https://github.com/pgbouncer/pgbouncer/pull/1091).

I don't think it's worth listing all of these as separate items. I think most of these fit well under the "Multiple documentation and CI improvements." line. Even linking to all the PRs doesn't seem super useful either.

To be clear I do think it's important to show appreciation for people contributing these improvements.

**Comment by eulerto** [comment]
> > * We used to have **Features**, **Fixes** and **Cleanups** as sections. We recently introduced **Deprecations** and **Changes**. It seems the goal of Changes is to break Features into Important Features and Regular Features. It can cause some confusion IMO and I prefer to have only one section: **Features**. I'm not sure about this new section (Breaking Changes), I would put it into Features with the same proposed sentences. The "This is a minor breaking change" is clear that this commit introduces a different behavior.
> 
> I think people easily skip over a release note item description if after the first sentence they think it doesn't impact them. So I proposed to call it out a bit more heavily, especially since I don't think we make breaking changes often/ever. Keeping it in features and making **minor breaking change** bold would also be fine by me.
> 

The idea is to reduce the number of sections. I also think that we should merge Features and Changes.

> > * various commits are not mentioned in the release notes. It should be in Fixes or Cleanups such as [#1035 - patch to indicate pem certificates / keys are usable by pgbouncer #1036](https://github.com/pgbouncer/pgbouncer/pull/1036) [Fix typos #1045](https://github.com/pgbouncer/pgbouncer/pull/1045) [Silence deprecation warnings of c-ares #1046](https://github.com/pgbouncer/pgbouncer/pull/1046) [Fix tests on non US datestyle configured systems #1047](https://github.com/pgbouncer/pgbouncer/pull/1047) [Improves README #1051](https://github.com/pgbouncer/pgbouncer/pull/1051) [The 1st argument of calloc should be 'nelem'. #1073](https://github.com/pgbouncer/pgbouncer/pull/1073) [Fix minor details in config.md #1082](https://github.com/pgbouncer/pgbouncer/pull/1082) [Improve documentation on connecting to AWS RDS and SCRAM #1083](https://github.com/pgbouncer/pgbouncer/pull/1083) [Improve run python function #1091](https://github.com/pgbouncer/pgbouncer/pull/1091).
> 
> I don't think it's worth listing all of these as separate items. I think most of these fit well under the "Multiple documentation and CI improvements." line. Even linking to all the PRs doesn't seem super useful either.
> 
> To be clear I do think it's important to show appreciation for people contributing these improvements. That's what I try to do on the PRs always with a "Thank you". And listing the new contributors in the new contributors section on the Github release page: https://github.com/pgbouncer/pgbouncer/releases/tag/pgbouncer_1_22_0 (There's a button for github to automatically create that section).

WFM.

**Comment by emelsimsek** [comment]
@eulerto @JelteF I believe I have covered all your feedback. Note that I will try to add new contributors during the release process.

**Review by eulerto — APPROVED** [review]

**Review by JelteF — APPROVED** [review]

**Pull request merged**